benefits/enrollment_switchio/api.py

```diff
@@ -133,25 +165,113 @@ def get_registration_status(self, registration_id, timeout=5) -> RegistrationSta
 
         return RegistrationStatus(**response.json())
 
-    # see https://github.com/cal-itp/benefits/issues/2848 for more context about this
-    def _cert_request(self, request_func):
-        """
-        Creates named (on-disk) temp files for client cert auth.
-        * request_func: curried callable from `requests` library (e.g. `requests.get`).
-        """
-        # requests library reads temp files from file path
-        # The "with" context destroys temp files when response comes back
-        with NamedTemporaryFile("w+") as cert, NamedTemporaryFile("w+") as key, NamedTemporaryFile("w+") as ca:
-            # write client cert data to temp files
-            # resetting so they can be read again by requests
-            cert.write(self.client_certificate)
-            cert.seek(0)
 
-            key.write(self.private_key)
-            key.seek(0)
+@dataclass
+class Group:
+    id: int
+    operatorId: int
+    name: str
+    code: str
+    value: int
 
-            ca.write(self.ca_certificate)
-            ca.seek(0)
 
-            # request using temp file paths
-            return request_func(verify=ca.name, cert=(cert.name, key.name))
+class EnrollmentClient(Client):
+
+    def __init__(self, api_url, authorization_header_value, private_key, client_certificate, ca_certificate):
+        super().__init__(private_key, client_certificate, ca_certificate)
+        self.api_url = api_url
+        self.authorization_header_value = authorization_header_value
+
+    def _get_headers(self):
+        return {"Authorization": self.authorization_header_value}
+
+    def healthcheck(self, timeout=5):
+        request_path = "/api/external/discount/echo"
+
+        response = self._cert_request(
+            lambda verify, cert: requests.get(
+                self.api_url.strip("/") + request_path,
```

Comment:
This `.strip("/")` could be done once when `self.api_url` is set, to simplify later usage.